modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/AddFormInstanceRecordMVCActionCommand.java

```diff
@@ -149,11 +149,15 @@ protected void doProcessAction(
 		String redirectURL = ParamUtil.getString(
 			actionRequest, "redirect", ddmFormInstanceSettings.redirectURL());
 
+		String currentURL = ParamUtil.getString(actionRequest, "currentURL");
+
 		DDMFormSuccessPageSettings ddmFormSuccessPageSettings =
 			ddmForm.getDDMFormSuccessPageSettings();
 
-		if (Validator.isNotNull(redirectURL) ||
+		if ((Validator.isNotNull(redirectURL) &&
+			 !StringUtil.equals(currentURL, redirectURL)) ||
```

Comment:
Could you please give more details on why do you need to verify this?

Reply:
Hi @rodrigopaulino . Please imagine a scenario where the user is creating a Form that he sets a Redirect URL and he doesn't have a success page in his form. This condition allows that when returning to the form, the toast message is not thrown on the screen, whether the form is created with or without a success page.

Reply:
Thanks for explaining, @rijema. What I don't understand is when the currentURL will be equals to the redirectURL. Do you have an example of that?

Reply:
When we don't set an explicit redirectURL, the code interprets that we're doing a redirection to the same form. In this case, the redirectURL will be the same as the current. Does it make sense to you, @rodrigopaulino?

Reply:
Does that mean that there's always going to be redirectURL?

Reply:
Wouldn't it be easier to just check if the setting exists?

Reply:
#`Does that mean that there's always going to be redirectURL?`

Yes. 

#Wouldn't it be easier to just check if the redirectURL setting exists?

I don't think so, since the object will always be filled with an URL, according to the way it was created. I mean, in my testings with different forms I could not found a single use case the value would be empty. But please tell me if I'm being too conclusive in defining this. I can do some more testing with other use cases.

Reply:
Hi, @rijema. Thanks for explaining your thought process. I trust your investigation and judgement. I just want to understand the scenarios in order to come up with the best solution. Now, here's what I'm thinking...
If the redirectURL is never null, why are we checking [this](https://github.com/liferay-forms/liferay-portal/pull/1961/files#diff-d6f8076fb627c946e6e93173fb0dfe74cd724f71a17b99263707b90f5d5e77e4R152) then? Also, I think I wasn't clear before when I said "redirectURL setting". What I meant by that was `ddmFormInstanceSettings.redirectURL()`. Wouldn't it be easier to simply verify if there's a `ddmFormInstanceSettings.redirectURL()` (the one that you type and set inside the form settings modal) to hide the default success message?